ui/src/app/views/app/component.ts

```diff
@@ -97,6 +108,50 @@ export class AppComponent implements OnInit {
     });
   }
 
+  /**
+   * On Navigation start, show a spinner.  On navigation end, stop the spinner.
+   */
+  private spinOnNavigate(event: RouterEvent): void {
+    const spinner = this.pageSpinner.nativeElement;
+    const spinParent = this.renderer.parentNode(spinner);
+
+    if (event instanceof NavigationStart) {
+      this.zone.runOutsideAngular(() => {
+        this.renderer.setStyle(spinParent, 'opacity', 0.8);
+        this.renderer.setStyle(spinParent, 'z-index', 1);
+        this.renderer.setStyle(spinner, 'opacity', 1);
+      });
+    } else if (event instanceof NavigationEnd
+            || event instanceof NavigationCancel
+            || event instanceof NavigationError) {
+      this.zone.runOutsideAngular(() => {
+        setTimeout(() => {
+        this.renderer.setStyle(spinParent, 'opacity', 0);
+        this.renderer.setStyle(spinParent, 'z-index', -1);
+        this.renderer.setStyle(spinner, 'opacity', 0);
+        }, 500);
```

Comment:
Why 500ms here? If another NavigationStart came in during this window we'd get undesirable behavior.

Reply:
Yes - completely agree we need a spinner and it would be nice if it didn't flash briefly. My comment was only about this 500ms delay before removal. If this is for debugging, then it would be good to remove it before merging this PR, or adding a TODO to indicate it is temporary. It could lead to buggy behavior as currently implemented.